app/controllers/warehouse_reports/youth_intake_export_controller.rb

```diff
@@ -36,10 +36,32 @@ def create
         preload(:client, :youth_intakes).
         visible_by?(current_user).
         between(start_date: @filter.start, end_date: @filter.end)
+      @housing_resolution_plans = GrdaWarehouse::Youth::HousingResolutionPlan.ordered.
+        joins(:client).
+        preload(:client, :youth_intakes).
+        visible_by?(current_user).
+        between(start_date: @filter.start, end_date: @filter.end)
+
+      commit = params[:commit]
       respond_to do |format|
         format.xlsx do
-          filename = "Youth Intake Export #{Time.current.to_s.delete(',')}.xlsx"
-          render(xlsx: 'index', filename: filename)
+          case commit
+          when 'Download Data'
+            xlsx_filename = "Youth Intake Export #{Time.current.to_s.delete(',')}.xlsx"
+            render(xlsx: 'index', filename: xlsx_filename)
+          when 'Download Per-Client Data'
```

Comment:
I'm a little worried this might not complete in the allotted time.  Do you have a sense of how fast this is for 100+ clients?